trie/trie_test.go

```diff
@@ -0,0 +1,255 @@
+
+		found, matched := trie.Get("hello world")
+		require.True(t, found)
+		require.Equal(t, "hello", matched)
+
+		found, matched = trie.Get("world")
+		require.False(t, found)
+		require.Equal(t, "", matched)
+
+		// number and special characters
+		trie.Insert("1234567890")
+		trie.Insert("!@#$%^&*()")
+
+		found, matched = trie.Get("1234567890")
+		require.True(t, found)
+		require.Equal(t, "1234567890", matched)
+
+		found, matched = trie.Get("!@#$%^&*()1234567890")
+		require.True(t, found)
+		require.Equal(t, "!@#$%^&*()", matched)
+
+		found, matched = trie.Get("1234567890!@#$%^&*()")
+		require.True(t, found)
+		require.Equal(t, "1234567890", matched)
+	})
+
+	t.Run("non-ascii", func(t *testing.T) {
+		trie := NewTrie()
+
+		trie.Insert("你好")
+		trie.Insert("世界 ABCD")
+
+		found, matched := trie.Get("你好世界")
+		require.True(t, found)
+		require.Equal(t, "你好", matched)
+
+		found, matched = trie.Get("你")
+		require.False(t, found)
+		require.Equal(t, "", matched)
+
+		found, matched = trie.Get("世界 ABCD")
+		require.True(t, found)
+		require.Equal(t, "世界 ABCD", matched)
+	})
+
+	t.Run("empty string", func(t *testing.T) {
+		trie := NewTrie()
+		trie.Insert("")
+
+		// Test querying with an empty string. Even if "" was inserted,
+		// it shouldn't be considered a valid match.
+		found, matched := trie.Get("")
+		require.False(t, found)
+		require.Equal(t, "", matched)
+	})
+
+	t.Run("spaces and tabs", func(t *testing.T) {
+		trie := NewTrie()
+		trie.Insert(" ")
+
+		found, matched := trie.Get(" abc")
+		require.True(t, found)
+		require.Equal(t, " ", matched)
+
+		found, matched = trie.Get("  ")
+		require.True(t, found)
+		require.Equal(t, " ", matched)
+
+		found, matched = trie.Get("abc")
+		require.False(t, found)
+		require.Equal(t, "", matched)
+
+		// tab
+		trie.Insert("	")
+
+		found, matched = trie.Get("	 abc")
+		require.True(t, found)
+		require.Equal(t, "	", matched)
+
+		found, matched = trie.Get("	")
+		require.True(t, found)
+		require.Equal(t, "	", matched)
+
+		found, matched = trie.Get("abc")
+		require.False(t, found)
+		require.Equal(t, "", matched)
+	})
+
+	t.Run("multiple word insertions with common prefixes", func(t *testing.T) {
+		trie := NewTrie()
+		trie.Insert("ab")
+		trie.Insert("abc")
+
+		found, matched := trie.Get("abcd")
+		require.True(t, found)
+		require.Equal(t, "ab", matched)
+
+		found, matched = trie.Get("ab")
+		require.True(t, found)
+		require.Equal(t, "ab", matched)
+
+		trie = NewTrie()
+		trie.Insert("a")
+		trie.Insert("ab")
+		trie.Insert("abc")
+
+		found, matched = trie.Get("abcd")
+		require.True(t, found)
+		require.Equal(t, "a", matched)
+
+		found, matched = trie.Get("abd")
+		require.True(t, found)
+		require.Equal(t, "a", matched)
+
+		found, matched = trie.Get("ad")
+		require.True(t, found)
+		require.Equal(t, "a", matched)
+
+		found, matched = trie.Get("b")
+		require.False(t, found)
+		require.Equal(t, "", matched)
+	})
+
+	t.Run("case sensitivity", func(t *testing.T) {
```

Comment:
There is no mention in the Trie documentation that it is case sensible. Please add it.